p2p/pex/pex_reactor.go

```diff
@@ -120,11 +121,11 @@ func (r *PEXReactor) OnStart() error {
 		return err
 	}
 
-	// return err if user provided a bad seed address
-	// or a host name that we cant resolve
-	seedAddrs, err := r.checkSeeds()
+	numOnline, seedAddrs, err := r.checkSeeds()
 	if err != nil {
 		return err
+	} else if numOnline == 0 && r.book.Empty() {
+		return errors.New("Address book is empty, and could not connect to any seed nodes")
```

Comment:
This behaviour makes sense to me. If no seed node was online, it may take awhile for an operator to notice. If the seed node wasn't reachable the first ping, you likely want to configure your seed node listing / make sure you're connected to the internet.

I guess we could try connecting to the seeds like 3 times, but I definitely don't think we should keep trying indefinitely long.